[DEVI-1570] Updates PagerDuty App docs for Scoped OAuth (#68)
PagerDuty/developer-docs: docs/app-integration-development/08-OAuth-Functionality.md

PagerDuty Apps with Classic User OAuth can be published for use with other PagerDuty accounts.

### Scoped OAuth

Scoped OAuth allows an application to act as the PagerDuty App itself by obtaining an app token using an OAuth 2.0 client credentials flow. Optionally, it can also act as a user after obtaining their authorization and consent. See [Obtaining an App OAuth Token](12-App-OAuth-Token.md), [Obtaining a User OAuth Token via Code Grant](09-User-OAuth-Token-Code-Grant.md) and [Obtaining a User OAuth Token via PKCE](10-User-OAuth-Token-PKCE.md) for the details of obtaining each type of token.

The access available to an application using Scoped OAuth differs by token type:
* With an app token, access is only limited by the scopes granted to the application. If the application has the `incidents.read` scope, it will be able to read all incidents on the account.

> * With a user token, access is the intersection of the scopes granted to the application and the permissions of the user. If an application has the `incidents.read` scope and is acting as user Pagey, it will only be able to read incidents that user Pagey has permission to see.

Reviewer: ```suggestion
* With a user token, access is the intersection of the scopes granted to the application and the permissions of the user. If an application has the `incidents.read` scope and was authorized by a user named Pagey, it will only be able to read incidents that Pagey has permission to see.
```
